subsys/net/lib/lwm2m/lwm2m_engine.c

```diff
@@ -2258,6 +2258,12 @@ static int lwm2m_read_handler(struct lwm2m_engine_obj_inst *obj_inst,
 			continue;
 		}
 
+		/* v1.1 allows resource instance level access */
+		if (msg->path.level == 4U &&
```

Comment:
I wonder if that should be conditionally executed (`if (IS_ENABLED(CONFIG_LWM2M_VERSION_1_1) && ...`) once the PR adding version gets in, same for others.

Reply:
That's something I was also pondering but then I read the v1.0.2 in a way that resource instance-level access is unspecified - not forbidden. It might be that I'm a bit too creative with my interpretation...

Reply:
Fixed.